p2p/src/network/kad/bootstrap/p2p_network_kad_bootstrap_reducer.rs

```diff
@@ -111,29 +154,45 @@ impl P2pNetworkKadBootstrapState {
                             closest_peers: closest_peers.clone(),
                         },
                     );
+                    Ok(())
                 } else {
-                    return Err(format!("cannot find stats for request {req:?}"));
+                    Err(format!("cannot find stats for request {req:?}"))
+                };
+
+                if state.successful_requests < 20 {
+                    let dispatcher = state_context.into_dispatcher();
+                    dispatcher.push(P2pNetworkKadBootstrapAction::CreateRequests);
                 }
 
-                Ok(())
+                result
             }
             P2pNetworkKadBootstrapAction::RequestError { peer_id, error } => {
-                let Some(req) = self.requests.remove(peer_id) else {
-                    return Err(format!("cannot find reques for peer {peer_id}"));
+                let bootstrap_state: &mut P2pNetworkKadBootstrapState =
+                    state_context.get_substate_mut()?;
+
+                let Some(req) = bootstrap_state.requests.remove(peer_id) else {
```

Comment:
Same as above, lets remove the interleaving of state-updates and control flow logic